### PR TITLE
Feature/update intellij plugin

### PR DESCRIPTION
**Pull request by fudler**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -26,22 +27,36 @@ tasks {
...
+        instrumentationTools()
+
+        testFramework(TestFrameworkType.Plugin.Java)
+    }
+
     implementation(kotlin("stdlib", kotlinVersion))
 
     testImplementation(kotlin("stdlib-jdk7", kotlinVersion))
-    testImplementation("com.tngtech.jgiven:jgiven-junit:1.2.5")
+    testImplementation("com.tngtech.jgiven:jgiven-junit:1.3.0")
```
Comment (l-1squared) [inline build.gradle.kts:54]: Don't we have renovat here :/

File: example/build.gradle [hunk 2]
```diff
@@ -5,6 +5,6 @@ repositories {
 }
 
 dependencies {
-    testImplementation 'com.tngtech.jgiven:jgiven-junit:1.2.0'
+    testImplementation 'com.tngtech.jgiven:jgiven-junit:1.3.0'
     testImplementation 'junit:junit:4.13.2'
 }
```
Comment (l-1squared) [inline example/build.gradle:10]: POSIX file ending

File: build.gradle.kts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
 val jetbrainsPublishUsername: String? by project
 val jetbrainsPublishToken: String? by project
 
 plugins {
-    id("org.jetbrains.intellij") version "1.15.0"
-    kotlin("jvm") version "1.9.0"
+    id("org.jetbrains.intellij.platform") version "2.1.0"
+    id("org.jetbrains.kotlin.jvm") version "2.1.0-Beta1"
```
Comment (l-1squared) [inline build.gradle.kts:10]: do we HAVE to use Beta1?

File: build.gradle.kts (hunk 1 above)
Comment (fudler) [reply build.gradle.kts:54]: We have, but the stuff wasn't merged for ages. So we increase the version by hand at the moment and rely on renovate again later on.

File: build.gradle.kts (hunk 3 above)
Comment (fudler) [reply build.gradle.kts:10]: We can switch to a stable on for sure.

**Pull request merged**